python/dset/verification_dataset.py

```diff
@@ -0,0 +1,217 @@
+from bench.geom import get_E_matrix, get_F_matrix, to_pixel_coords, get_inliers_F
+dirname = os.path.dirname(os.path.abspath(__file__))
+
+class verification_dataset():
+
+    def __init__(self, data_names=['reichstag','brown_bm_3_05']):
+        """Verification dataset
+
+        Attributes
+        ----------
+        data_names: list
+            List of data sets to be used. By default loads 'reichstag' and 'brown_bm_3_05'
+            dataset.
+        """
+
+        self.name='verification_dset'
+        self.data_dump_prefix = os.path.join(dirname,"../../datasets/Verification/data_dump")
+        self.data_names = data_names
+
+        self._load_data()
+        self.sequences = list(self.data.keys())
+        self._prepare_data()
+
+
+    def _load_data(self, var_mode='test'):
+
+        """Main data loading routine adapted from
+        https://github.com/vcg-uvic/learned-correspondence-release
+
+        Loads data defined in @var_name_list from files at @data_dump_prefix and @data_names
+        into @self.data
+
+        Attributes
+        ----------
+        var_mode: str
+            what split to us in the dataset. Default is 'test' but 'train' or val can be
+            used as well.
+        """
+
+        print("Loading {} data".format(var_mode))
+
+        # use only the first two characters for shorter abbrv
+        var_mode = var_mode[:2]
+
+        # Now load data.
+        var_name_list = [
+            "xs", "ys", "Rs", "ts",
+            "img1s", "cx1s", "cy1s", "f1s",
+            "img2s", "cx2s", "cy2s", "f2s", "k1s", "k2s",
+        ]
+
+        data_folder = self.data_dump_prefix
+
+        # Let's unpickle and save data
+        data = {}
+
+        for data_name in self.data_names:
+            cur_data_folder = "/".join([
+                data_folder,
+                data_name,
+                "numkp-2000",
+                "nn-1",
+                "nocrop",
+            ])
+
+
+            suffix = "{}-1000".format(var_mode)
+
+            cur_folder = os.path.join(cur_data_folder, suffix)
+            ready_file = os.path.join(cur_folder, "ready")
+            if not os.path.exists(ready_file):
+                # data_gen_lock.unlock()
+                raise RuntimeError("Data is not prepared!")
+
+            for var_name in var_name_list:
+                cur_var_name = var_name + "_" + var_mode
+                in_file_name = os.path.join(cur_folder, cur_var_name) + ".pkl"
+                with open(in_file_name, "rb") as ifp:
+                    if data_name in data:
+                        if var_name in data[data_name]:
+                            data[data_name][var_name] += pickle.load(ifp)
+                        else:
+                            data[data_name][var_name] = pickle.load(ifp)
+                    else:
+                        data[data_name] = {}
+                        data[data_name][var_name] = pickle.load(ifp)
+
+        self.data = data
+
+    def _prepare_data(self):
```

Comment:
function is a bit long currently, can we make it shorter? one simple way  would be to combine lines, e.g.

`coords1, coords2 = list(), list()`